src/uct/ib/dc/dc_mlx5_devx.c

```diff
@@ -50,7 +52,16 @@ ucs_status_t uct_dc_mlx5_iface_devx_create_dct(uct_dc_mlx5_iface_t *iface)
     if (!uct_ib_iface_is_roce(&iface->super.super.super)) {
         UCT_IB_MLX5DV_SET(dctc, dctc, pkey_index, ib_iface->pkey_index);
     }
-    UCT_IB_MLX5DV_SET(dctc, dctc, port, ib_iface->config.port_num);
+
+    if (iface->rx.port_affinity == 0) {
+        port_affinity = ib_iface->config.port_num;
+    } else if (iface->rx.port_affinity == (uint8_t)(-1)) {
+        ucs_rand_range(1, dev->lag_level, &port_affinity);
+    } else {
+        port_affinity = iface->rx.port_affinity;
+    }
```

Comment:
we should 
we can initialize iface->rx.port_affinity to the specific affinity value already in uct_dc_mlx5_iface_init_port_affinity(), so the logic in this function will be fully moved to uct_dc_mlx5_iface_init_port_affinity(), and here we only do
`UCT_IB_MLX5DV_SET(dctc, dctc, port, iface->rx.port_affinity);`
this should simplify the overall logic